### ISSUE TITLE
Wrong Error Message "Add is not a Trait"

### ISSUE THREAD

**Issue by CryZe** [rust-lang/rust#35987]
I made a mistake while implementing Add for a type. Instead of using `+` for specifying multiple constraints, I accidentally used `,` and for some reason the compiler can't handle this very well. Instead of complaining about the `,` it says that `Add is not a Trait`

[Playground Link](https://play.rust-lang.org/?gist=fa1cbdc67f59ea33616f121ae697654a&version=nightly&backtrace=0)

``` rust
struct Okok<T: Clone>(T);

use std::ops::Add;

impl<T: Clone, Add> Add for Okok<T> {
    type Output = usize;

    fn add(self, rhs: Self) -> Self::Output {
        unimplemented!();
    }
}
...
```

```
error[E0404]: `Add` is not a trait
 --> <anon>:5:21
  |
5 | impl<T: Clone, Add> Add for Okok<T> {
  |                     ^^^ not a trait

error: cannot continue compilation due to previous error
```

This happens on both Nightly, Beta and Stable.

**Comment by petrochenkov** [comment]
Compiler can't complain about `,`, everything is syntactically correct - the first `Add` is interpreted as a fresh type parameter.

The error message should say something like "error: expected trait, found type parameter `Add`" + "note: ^^^ type parameter `Add` is defined here"

**Comment by CryZe** [comment]
Oh yeah, that would make sense.
